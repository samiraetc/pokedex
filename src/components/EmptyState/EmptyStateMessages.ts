
export const EmptyStateMessage = {
    EMPTY_STATE_MESSAGE: `You don't have any favorites pokemon yet`
}