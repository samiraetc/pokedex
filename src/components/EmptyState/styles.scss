.pokedex-empty-state {
    position: absolute;
    top: 55%;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    gap: $pokedex-spacing-20;
    align-items: center;

    &__icon {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        width: 200px;
    }

    &__title {
        color: $pokedex-color-blue-strong;
        font-size: $pokedex-font-size-16;
    }
}

