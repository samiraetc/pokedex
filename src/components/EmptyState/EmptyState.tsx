import image from '../../assets/gifs/no-favorites.gif'
import { EmptyStateMessage } from './EmptyStateMessages';

export const COMPONENT_ID = 'pokedex-empty-state';

const EmptyState = () => {
    const wordings = EmptyStateMessage;
    return (
        <div id={COMPONENT_ID} className={COMPONENT_ID}>
            <img src={image} alt="pokemon icon" className='pokedex-empty-state__icon' />
            <h1 className='pokedex-empty-state__title' >{wordings.EMPTY_STATE_MESSAGE}</h1>
        </div>
    )
}

export default EmptyState;