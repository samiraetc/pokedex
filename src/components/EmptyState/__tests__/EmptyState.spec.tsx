import { render, configure, fireEvent } from '@testing-library/react'
import EmptyState, { COMPONENT_ID } from '../EmptyState';
import { EmptyStateMessage } from '../EmptyStateMessages';

configure({
    testIdAttribute: 'id',
  });
  
describe('[EMPTY-STATE]', () => {
    const wordings = EmptyStateMessage;
    test('should render Empty State', () => {
        const { asFragment, getByTestId, getByText} = render(<EmptyState />);

        expect(getByText(wordings.EMPTY_STATE_MESSAGE))
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});