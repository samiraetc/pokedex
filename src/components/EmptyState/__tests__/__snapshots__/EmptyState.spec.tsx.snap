// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EMPTY-STATE] should render Empty State 1`] = `
<DocumentFragment>
  <div
    class="pokedex-empty-state"
    id="pokedex-empty-state"
  >
    <img
      alt="pokemon icon"
      class="pokedex-empty-state__icon"
      src="default"
    />
    <h1
      class="pokedex-empty-state__title"
    >
      You don't have any favorites pokemon yet
    </h1>
  </div>
</DocumentFragment>
`;
